pkg/env/env.go

```diff
@@ -20,6 +20,14 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/refreshable"
 )
 
+type EnvironmentType uint8
+
+const (
+	EnvironmentTypeProduction EnvironmentType = 1 << iota
```

Comment:
Left shift? Why `1 << iota` instead of just `iota`?

Edit: ahh, because it allows for bitmask matching.

Vague questions: is this worth the complexity it adds? Would we ever have multiple environment flags set? We don't really do masking or bitwise operations anywhere else in this codebase, so I'd be worried that the terseness this provides is at the expense of posing a barrier to entry/quick understanding.

Reply:
This added complexity is shifted from the naming of things that are "only in dev" or "only in prod", since commonly (always, right now?) we will do something in prod + integration, not just prod. So, when there's the `OnlyInEnv` runsteps Step, it can take a bit mask like so:

```
steps.OnlyInEnv(i.env, env.EnvironmentTypeProduction | env.EnvironmentTypeIntegration, steps.Action(i.doSomethingNotNeededInDev))
```

The thought here is that it's then more explicit where we're checking for the environment type and quicker to understand there.